geonode/base/api/serializers.py

```diff
@@ -485,7 +484,12 @@ def __init__(self, *args, **kwargs):
 
         self.fields["favorite"] = FavoriteField(read_only=True)
 
-    metadata = DynamicRelationField(ExtraMetadataSerializer, embed=False, many=True, deferred=True)
+    # Load metadata_records for contrib apps
+    if getattr(settings, "EXTRA_METADATA_ENABLED", False):
```

Comment:
Hi @t-book i would evaluate setting the variable as follow, by providing the default argument it will be always false if not declared

```python
deferred = getattr(settings, "EXTRA_METADATA_ENABLED", False)
```